staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -675,6 +595,14 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 		if !errors.IsNotFound(err) {
 			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%s\nfrom server for:", info.String()), info.Source, err)
 		}
+		// If the object doesn't exist and serverSideAuto is set
+		// server-side apply
+		if o.ServerSideApplyStrategy == cmdutil.ServerSideAuto {
+			return o.serverSideApplyOneObject(info, helper, originalData)
+		}
+
+		//set the field manager now that we have eliminated ServerSideAuto
+		helper = helper.WithFieldManager(o.FieldManager)
```

Comment:
I don't think this will result in the correct field manager--`o.FieldManager` never contains the default, and we want the CSA default here when it is unset. Let's make sure we have regression coverage for this.